RecoTracker/LSTCore/src/alpaka/PT2.h

```diff
@@ -0,0 +1,343 @@
+      auto const gridThreadExtent = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+
+      for (unsigned int i_pLS = globalThreadIdx[1]; i_pLS < nPixelSegments; i_pLS += gridThreadExtent[1]) {
+        auto iLSModule_max = connectedPixelIndex[i_pLS] + connectedPixelSize[i_pLS];
+
+        for (unsigned int iLSModule = connectedPixelIndex[i_pLS] + globalBlockIdx[0]; iLSModule < iLSModule_max;
+             iLSModule += gridBlockExtent[0]) {
+          uint16_t segmentLowerModuleIndex =
+              modulesInGPU
+                  .connectedPixels[iLSModule];  //connected pixels will have the appropriate lower module index by default!
+#ifdef WARNINGS
+          if (tripletLowerModuleIndex >= *modulesInGPU.nLowerModules) {
+            printf("tripletLowerModuleIndex %d >= modulesInGPU.nLowerModules %d \n",
+                   tripletLowerModuleIndex,
+                   *modulesInGPU.nLowerModules);
+            continue;  //sanity check
+          }
+#endif
+          //Removes 2S-2S :FIXME: filter these out in the pixel map
+          if (modulesInGPU.moduleType[segmentLowerModuleIndex] == lst::TwoS)
+            continue;
+
+          uint16_t pixelModuleIndex = *modulesInGPU.nLowerModules;
+          unsigned int nOuterSegments = segmentsInGPU.nSegments[segmentLowerModuleIndex];
+          if (nOuterSegments == 0)
+            continue;
+
+          unsigned int pixelSegmentIndex = rangesInGPU.segmentModuleIndices[pixelModuleIndex] + i_pLS;
+
+          if (segmentsInGPU.isDup[i_pLS])
+            continue;
+          if (segmentsInGPU.partOfPT5[i_pLS])
+            continue;  //don't make pT3s for those pixels that are part of pT5
+          if (segmentsInGPU.partOfPT3[i_pLS])
+            continue; 
+
+          short layer2_adjustment;
+          if (modulesInGPU.layers[segmentLowerModuleIndex] == 1) {
+            layer2_adjustment = 1;
+          }  //get upper segment to be in second layer
+          else if (modulesInGPU.layers[segmentLowerModuleIndex] == 2) {
+            layer2_adjustment = 0;
+          }  // get lower segment to be in second layer
+          else {
+            continue;
+          }
+
+          //fetch the segment
+          for (unsigned int outerSegmentArrayIndex = globalThreadIdx[2]; outerSegmentArrayIndex < nOuterSegments;
+               outerSegmentArrayIndex += gridThreadExtent[2]) {
+            unsigned int outerSegmentIndex =
+                rangesInGPU.segmentModuleIndices[segmentLowerModuleIndex] + outerSegmentArrayIndex;
+            if (modulesInGPU.moduleType[segmentsInGPU.outerLowerModuleIndices[outerSegmentIndex]] == lst::TwoS)
+              continue;  //REMOVES PS-2S
+
+            if (segmentsInGPU.partOfPT5[outerSegmentIndex])
+              continue;  //don't create pT2s for T2s accounted in pT5s
+
+            if (segmentsInGPU.partOfPT3[outerSegmentIndex])
+              continue; //don't create pT2s for T2s accounted in pT3s
```

Comment:
this is probably the origin of the crashes: `partOf*` variables are in the pLS sub-SoA, not the OT LS.
This should be removed. If segments (OT LS) used somewhere else should be checked for use, new variables should be added to the OT part of the Segments SoA

Reply:
I added in the pT3 kernels and pT5 kernels, and check if the segment is used in their buildings. I think these lines are filling the partOfPT3 variable.
https://github.com/SegmentLinking/cmssw/blob/f8d0fc0404184ce17f8101c5e477c01374f553a6/RecoTracker/LSTCore/src/alpaka/PixelTriplet.h#L1020